plaso/storage/file_interface.py

```diff
@@ -0,0 +1,1147 @@
+# -*- coding: utf-8 -*-
+"""Storage interface classes for file-backed stores."""
+
+from __future__ import unicode_literals
+
+import shutil
+
+import abc
+import os
+import tempfile
+
+from plaso.lib import definitions
+from plaso.serializer import json_serializer
+from plaso.storage import interface
+
+class SerializedAttributeContainerList(object):
+  """Serialized attribute container list.
+
+  The list is unsorted and pops attribute containers in the same order as
+  pushed to preserve order.
+
+  The GetAttributeContainerByIndex method should be used to read attribute
+  containers from the list while it being filled.
+
+  Attributes:
+    data_size (int): total data size of the serialized attribute containers
+        on the list.
+    next_sequence_number (int): next attribute container sequence number.
+  """
+
+  def __init__(self):
+    """Initializes a serialized attribute container list."""
+    super(SerializedAttributeContainerList, self).__init__()
+    self._list = []
+    self.data_size = 0
+    self.next_sequence_number = 0
+
+  @property
+  def number_of_attribute_containers(self):
+    """int: number of serialized attribute containers on the list."""
+    return len(self._list)
+
+  def Empty(self):
+    """Empties the list."""
+    self._list = []
+    self.data_size = 0
+
+  def GetAttributeContainerByIndex(self, index):
+    """Retrieves a specific serialized attribute container from the list.
+
+    Args:
+      index (int): attribute container index.
+
+    Returns:
+      bytes: serialized attribute container data or None if not available.
+
+    Raises:
+      IndexError: if the index is less than zero.
+    """
+    if index < 0:
+      raise IndexError(
+          'Unsupported negative index value: {0:d}.'.format(index))
+
+    if index < len(self._list):
+      return self._list[index]
+
+    return None
```

Comment:
Why not raise for index > len() ?

Reply:
I don't know - this class was just moved.